docs/developers/contributing.rst

```diff
@@ -329,10 +329,16 @@ Release procedure
    Most of the release process is now handled by GitHub workflow which should
    automatically push a release to PyPI if a tag is pushed.
 
-Before releasing, make sure that all pull requests which will be
-included in the release have been properly documented in
-`docs/release.rst`.
-
+Pre-release
+"""""""""""
+1. Make sure that all pull requests which will be
+   included in the release have been properly documented in
+   :file:`docs/release-notes.rst`.
+2. Rename the "Unreleased" section heading in :file:`docs/release-notes.rst`
```

Comment:
That seems a bit annoying because it requires another commit before the release or am I understanding it wrong?

Reply:
Yeah, that's right. I guess with the current system, unless we decide on the version numbering ahead of time then this is unavoidable?

Reply:
On other projects I've used https://towncrier.readthedocs.io/en/stable/ with success, and it was suggested a while ago here too: https://github.com/zarr-developers/zarr-python/issues/829.

You still have to generate the changelog before release, but it makes having multiple branches easier (e.g., if we have a 3.0.x branch for bugfixes and `main` for 3.x.0 releases), and would mean the extra commit after release (your comment below) wouldn't be needed.

I'd be up for trying to set towncrier up, either instead of or after this PR?